pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -239,7 +240,12 @@ func (nm *NodeManager) addNode(node *v1.Node) {
 
 func (nm *NodeManager) removeNode(node *v1.Node) {
 	nm.registeredNodesLock.Lock()
+	nm.nodeInfoLock.Lock()
```

Comment:
I think this can create a deadlock. Can you please change this?

Reply:
I don't see any existing workflow that may hold a nodeInfoLock and meantime try to acquire registeredNodesLock, so deadlock should not really happen. I do agree with you it is not a good idea to acquires two locks here, but unless we make a refactoring to merge these two maps, I don't see an option to avoid this. Do you have any better suggestions?

Reply:
@shaominchen Currently we don't have any workflow where deadlock can happen, yet to make it same let's add common methods so that sequence to acquire and release lock remains same.

Reply:
As discussed offline, we agree it's better to keep these 2 locks independently for now (before we merge those 2 maps into 1). Will update the logic.